cpp/src/join/semi_join.cu

```diff
@@ -71,65 +90,67 @@ std::unique_ptr<rmm::device_uvector<cudf::size_type>> left_semi_anti_join(
   auto right_flattened_keys = right_flattened_tables.flattened_columns();
   auto left_flattened_keys  = left_flattened_tables.flattened_columns();
 
-  // Only care about existence, so we'll use an unordered map (other joins need a multimap)
-  using hash_table_type = concurrent_unordered_map<cudf::size_type, bool, row_hash, row_equality>;
+  // Create hash table.
+  auto hash_table = cuco::
+    static_map<hash_value_type, size_type, cuda::thread_scope_device, default_allocator<char>>{
```

Comment:
Just realized this PR depends on https://github.com/rapidsai/cudf/pull/9704 as well. We are not passing the input `stream` to `default_allocator` in this way and a [stream allocator adaptor](https://github.com/rapidsai/cudf/pull/9704/files#diff-fe28a4362c6d7a5d462be7c23b9d0a2aab1d8e61cc1a4515f863652c53f01f9fR42) should be used. I will ping Jake to merge the dependencies first.

Reply:
Yep and #9704 has another cuco dependency I believe.

Reply:
I think I've adapted the code correctly now, but it's probably worth another quick look from you. Then I think this should be good to go.